lightning/src/routing/scoring.rs

```diff
@@ -241,6 +241,62 @@ impl ReadableArgs<u64> for FixedPenaltyScorer {
 	}
 }
 
+/// [`Score`] implementation that uses a dynamic penalty.
+pub struct DynamicPenaltyScorer<F>
```

Comment:
I chose the name `DynamicPenaltyScorer` because this basically replicates the functionality of `FixedPenaltyScorer`, but allows the user to define the behaviour dynamically by providing a closure.
Not sure though if this is the most appropriate naming, maybe `CustomPenaltyScorer` would be better?

Reply:
Alright, I added the `(C-not exported)` comment with c53a6c216f2fdbf68f46d9afcdef236ea4c232ac.

Reply:
> Actually, on second throught, can we just drop the struct and `impl <F: ...> Score for F`? Then we still get the `Score` but no need to bother with a wrapper struct (still needs the `(C-not exported)`)

I'm not sure I fully understand what you are suggesting? Could you reiterate?

Reply:
Seems like `DynamicPenaltyScorer` is a roundabout way of breaking `Score` into two traits. i.e. `find_route` doesn't need `payment_path_failed` and `payment_path_successful`.

Reply:
Dropped `DynamicPenaltyScorer` with https://github.com/lightningdevkit/rust-lightning/pull/1491/commits/19ecb4c02798caf3d89fd42e9497c78565efe756.